core/src/operator.rs

```diff
@@ -329,49 +329,66 @@ where
         }
     }
 
-    /// Checks if utxo is valid, spendable by operator and not spent
-    /// Saves the utxo to the db
-    #[tracing::instrument(skip(self), err(level = tracing::Level::ERROR), ret(level = tracing::Level::TRACE))]
+    /// Saves funding UTXO to the database.
     async fn set_funding_utxo(&self, funding_utxo: UTXO) -> Result<(), BridgeError> {
         self.db.set_funding_utxo(None, funding_utxo).await
     }
 
-    #[tracing::instrument(skip(self), err(level = tracing::Level::ERROR), ret(level = tracing::Level::TRACE))]
-    async fn is_profitable(
-        &self,
-        input_amount: u64,
-        withdrawal_amount: u64,
-    ) -> Result<bool, BridgeError> {
-        // Check if the withdrawal amount is within the acceptable range
-        if (withdrawal_amount - input_amount) > self.config.bridge_amount_sats {
-            Ok(false)
-        } else {
-            // Calculate net profit after the withdrawal
-            let net_profit = self.config.bridge_amount_sats - withdrawal_amount;
-            Ok(net_profit > self.config.operator_withdrawal_fee_sats.unwrap())
+    /// Checks if the withdrawal amount is within the acceptable range.
+    ///
+    /// # Parameters
+    ///
+    /// - `input_amount`:
+    /// - `withdrawal_amount`:
+    fn is_profitable(&self, input_amount: u64, withdrawal_amount: u64) -> bool {
+        if withdrawal_amount.wrapping_sub(input_amount) > self.config.bridge_amount_sats {
+            return false;
         }
+
+        // Calculate net profit after the withdrawal.
+        let net_profit = self.config.bridge_amount_sats - withdrawal_amount;
+
+        // Net profit must be bigger than withdrawal fee.
+        net_profit > self.config.operator_withdrawal_fee_sats.unwrap()
     }
 
+    /// Creates a withdrawal transaction with given signature and sends it to
```

Comment:
Verifies that the withdrawal has been made on Citrea, verifies a given SIGHASH_SINGLE|ANYONECANPAY signature, checks if it is profitable and funds the withdrawal